### PR TITLE
Fixes #1217, fixes #1303, fixes #1304 adding WMTS support to Leaflet,…

### LINKED ISSUE #1217: Introduce support for WMTS in the Catalog plugin
Depends on:
 * [ ] https://github.com/geosolutions-it/MapStore2/issues/1303
 * [ ] https://github.com/geosolutions-it/MapStore2/issues/1304

### LINKED ISSUE #1303: WMTS support for Leaflet
Introduce WMTS support for leaflet maps

### LINKED ISSUE #1304: WMTS support for OL3

### PR DESCRIPTION
**Pull request by mbarto**
… Openlayers and Catalog

### CODE REVIEW

**Comment by simboss** [comment]
At last! 

Never loose hope...

File: web/client/test-resources/wmts/GetCapabilities-1.0.0.xml [hunk 1]
```diff
@@ -0,0 +1,1913 @@
...
+        <ows:Identifier>EPSG:900913:15</ows:Identifier>
+        <ScaleDenominator>17061.836668423246</ScaleDenominator>
+        <TopLeftCorner>-2.003750834E7 2.0037508E7</TopLeftCorner>
+        <TileWidth>256</TileWidth>
+        <TileHeight>256</TileHeight>
+        <MatrixWidth>32768</MatrixWidth>
+        <MatrixHeight>32768</MatrixHeight>
+      </TileMatrix>
+      <TileMatrix>
+
```
Comment (offtherailz) [inline web/client/test-resources/wmts/GetCapabilities-1.0.0.xml]: generic url instead?

File: web/client/test-resources/wmts/GetCapabilities-1.0.0.xml [hunk 2]
```diff
@@ -0,0 +1,1913 @@
...
+            <MinTileCol>7</MinTileCol>
+            <MaxTileCol>7</MaxTileCol>
+          </TileMatrixLimits>
+          <TileMatrixLimits>
+            <TileMatrix>EPSG:4326:3</TileMatrix>
+            <MinTileRow>5</MinTileRow>
+            <MaxTileRow>5</MaxTileRow>
+            <MinTileCol>14</MinTileCol>
+            <MaxTileCol>14</MaxTileCol>
+          </TileMatrixLimits>
```
Comment (offtherailz) [inline web/client/test-resources/wmts/GetCapabilities-1.0.0.xml]: Can we reduce the number of levels to have a smaller file ?

File: web/client/utils/ogc/WMTS.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+/**
```
Comment (offtherailz) [inline web/client/utils/ogc/WMTS.js]: Is this file needed anymore ?

File: web/client/components/map/leaflet/__tests__/Layer-test.jsx [hunk 4]
```diff
@@ -179,6 +180,56 @@ describe('Leaflet layer', () => {
...
 
+    it('creates a wmts layer for leaflet map', () => {
+        var options = {
+            "type": "wmts",
+            "visibility": true,
+            "name": "nurc:Arc_Sample",
+            "group": "Meteo",
+            "format": "image/png",
+            "tileMatrixSet": "EPSG:900913",
+            "url": "http://demo.geo-solutions.it/geoserver/gwc/service/wmts"
```
Comment (offtherailz) [inline web/client/components/map/leaflet/__tests__/Layer-test.jsx]: generic url

File: web/client/components/map/leaflet/__tests__/Layer-test.jsx [hunk 5]
```diff
@@ -179,6 +180,56 @@ describe('Leaflet layer', () => {
...
+
+    it('creates a wmts layer with multiple urls for leaflet map', () => {
+        var options = {
+            "type": "wmts",
+            "visibility": true,
+            "name": "nurc:Arc_Sample",
+            "group": "Meteo",
+            "format": "image/png",
+            "tileMatrixSet": "EPSG:900913",
+            "url": ["http://demo.geo-solutions.it/geoserver/gwc/service/wmts", "http://demo.geo-solutions.it/geoserver/gwc/service/wmts"]
```
Comment (offtherailz) [inline web/client/components/map/leaflet/__tests__/Layer-test.jsx]: generic url for tests

File: web/client/components/map/openlayers/__tests__/Layer-test.jsx [hunk 6]
```diff
@@ -148,6 +149,50 @@ describe('Openlayers layer', () => {
...
 
+    it('creates a wmts layer for openlayers map', () => {
+        var options = {
+            "type": "wmts",
+            "visibility": true,
+            "name": "nurc:Arc_Sample",
+            "group": "Meteo",
+            "format": "image/png",
+            "tileMatrixSet": "EPSG:900913",
+            "url": "http://demo.geo-solutions.it/geoserver/gwc/service/wmts"
```
Comment (offtherailz) [inline web/client/components/map/openlayers/__tests__/Layer-test.jsx]: generic url

File: web/client/components/map/openlayers/__tests__/Layer-test.jsx [hunk 7]
```diff
@@ -148,6 +149,50 @@ describe('Openlayers layer', () => {
...
+
+    it('creates a wmts layer with multiple urls for openlayers map', () => {
+        var options = {
+            "type": "wmts",
+            "visibility": true,
+            "name": "nurc:Arc_Sample",
+            "group": "Meteo",
+            "format": "image/png",
+            "tileMatrixSet": "EPSG:900913",
+            "url": ["http://demo.geo-solutions.it/geoserver/gwc/service/wmts", "http://demo.geo-solutions.it/geoserver/gwc/service/wmts"]
```
Comment (offtherailz) [inline web/client/components/map/openlayers/__tests__/Layer-test.jsx]: generic url for testing

File: web/client/test-resources/wmts/GetCapabilities-1.0.0.xml [hunk 8]
```diff
@@ -0,0 +1,1913 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (offtherailz) [inline web/client/test-resources/wmts/GetCapabilities-1.0.0.xml]: Can we reduce size of this files (e.g. less levels/layers)?

File: web/client/test-resources/wmts/GetCapabilities-1.0.0.xml [hunk 9]
```diff
@@ -0,0 +1,1913 @@
...
+
+<Capabilities xmlns="http://www.opengis.net/wmts/1.0" xmlns:ows="http://www.opengis.net/ows/1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:gml="http://www.opengis.net/gml" xsi:schemaLocation="http://www.opengis.net/wmts/1.0 http://schemas.opengis.net/wmts/1.0/wmtsGetCapabilities_response.xsd" version="1.0.0">
+  <ows:ServiceIdentification>
+    <ows:Title>Web Map Tile Service - GeoWebCache</ows:Title>
+    <ows:ServiceType>OGC WMTS</ows:ServiceType>
+    <ows:ServiceTypeVersion>1.0.0</ows:ServiceTypeVersion>
+  </ows:ServiceIdentification>
+  <ows:ServiceProvider>
+    <ows:ProviderName>http://demo.geo-solutions.it/geoserver/gwc/service/wmts</ows:ProviderName>
```
Comment (offtherailz) [inline web/client/test-resources/wmts/GetCapabilities-1.0.0.xml]: and we should replace all theese urls?
